native/src/seal/util/scalingvariant.cpp

```diff
@@ -36,6 +36,10 @@ namespace seal
                     plain.data(), plain.data() + coeff_count, get<0>(I), get<0>(I),
                     [&](uint64_t m, uint64_t c) -> uint64_t {
                         m = barrett_reduce_64(m, plain_modulus);
+                        if (plain_modulus > cipher_modulus)
+                        {
+                            m = barrett_reduce_64(m, cipher_modulus);
```

Comment:
Fixes a failing test when built with HEXL on SKX. I didn't see the any failing tests with `sub_plain_without_scaling_variant`, but changed there as well to be safe.

Reply:
I have fixed this internally. :)